include/vcpkg/base/json.h

```diff
@@ -213,6 +213,7 @@ namespace vcpkg::Json
 
         // replaces the value if the key is found, otherwise inserts a new
         // value.
+        Value& insert_or_replace(std::string key, std::string value);
```

Comment:
I thought we intentionally wanted callers to need to say Json::Value::whatever to make sure there was no confusion here?

Reply:
At least in my opinion, strings are not confusing. They're also _so_ common that adding this overload significantly simplifies the rest of the codebase.

Reply:
I think it's fine too, but since it looked like this had been a deliberate design choice before I just wanted to highlight it as a change in policy.

Reply:
it was not a deliberate design choice - the reason I did this was due to overload resolution BS, iirc.